File: Exercícios/Capitulo_2/Propostos/q10.py
copo1 = "laranja"
copo2 = "acerola"

print("Antes da troca:")
print(f"Copo1: {copo1}, copo2: {copo2}")

"""Permutação trocando os valores das variáveis ao mesmo tempo
durante a execução do código"""

copo1, copo2 = copo2, copo1 
# em python a permuta de valores é possível mesmo sem a presença de variavel auxiliar

print("Depois da troca: ")
print(f"Copo1: {copo1}, copo2: {copo2}")

File: Exercícios/Capitulo_2/Propostos/q10.cpp
#include <iostream>
using namespace std;

int main()
{
    string copo1 = "laranja", copo2 = "acerola";

    cout << "Antes da troca:\n";
    cout << "Copo1: " << copo1 << ", copo2: " << copo2;

    string aux = "";

    // em c++ precisa-se de uma variável auxiliar para realizar a permuta de valores
    aux = copo1;
    copo1 = copo2;
    copo2 = aux;

    cout << "\nDepois da troca: \n";
    cout << "Copo1: " << copo1 << ", copo2: " << copo2;

    return 0;
}